### PR TITLE
Fix comparison of upgrade metadata

### PR DESCRIPTION
**Pull request by pchila**
## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added an entry in `./changelog/fragments` using the [changelog tool](https://github.com/elastic/elastic-agent#changelog)
- [ ] I have added an integration test or an E2E test

## Author's Checklist

- [ ]

## Related issues

- 

## Questions to ask yourself

- How are we going to support this in production? 
- How are we going to measure its adoption? 
- How are we going to debug this?
- What are the metrics I should take care of?
- ...

### CODE REVIEW

**Review by michalpristas — APPROVED** [review]
let's get this in, this is blocking other PRs

File: internal/pkg/agent/application/dispatcher/dispatcher.go [hunk 1]
```diff
@@ -321,7 +321,7 @@ func reportNextScheduledUpgrade(input []fleetapi.Action, detailsSetter details.O
 
 	upgradeDetails := details.NewDetails(nextUpgrade.Version, details.StateScheduled, nextUpgrade.ID())
 	startTime, _ := nextUpgrade.StartTime()
-	upgradeDetails.Metadata.ScheduledAt = startTime
+	upgradeDetails.Metadata.ScheduledAt = &startTime
```
Comment (pchila) [inline internal/pkg/agent/application/dispatcher/dispatcher.go:324]: I am not 100% sure this is correct, we probably need to test startTime for zero value (checked the `StartTime()` function)

Comment (pchila) [reply internal/pkg/agent/application/dispatcher/dispatcher.go]: see 73a9031524ab62212f12eef2428c29ea620aa458

**Review by rdner — APPROVED** [review]

File: internal/pkg/agent/application/dispatcher/dispatcher.go [hunk 2]
```diff
@@ -320,8 +320,10 @@ func reportNextScheduledUpgrade(input []fleetapi.Action, detailsSetter details.O
 	}
 
 	upgradeDetails := details.NewDetails(nextUpgrade.Version, details.StateScheduled, nextUpgrade.ID())
-	startTime, _ := nextUpgrade.StartTime()
-	upgradeDetails.Metadata.ScheduledAt = startTime
+	startTime, err := nextUpgrade.StartTime()
```
Comment (michalpristas) [inline internal/pkg/agent/application/dispatcher/dispatcher.go:323]: we should not have an error here as we're guarding against this in a loop. 
in case we end up with an error here following lines are questionable
cc @ycombinator

File: internal/pkg/agent/application/dispatcher/dispatcher.go (hunk 1 above)
Comment (ycombinator) [reply internal/pkg/agent/application/dispatcher/dispatcher.go:324]: It built before because two PRs independently succeeded in CI.  So they got merged into `main`.  But after that, the changes in them collided and we got this failure.

To prevent this in the future, we need to enable the *Require branches to be up to date before merging branch* protection.
